### ISSUE TITLE
Possible floating point exception in UpdateMouseInputs()

### ISSUE THREAD

**Issue by dpethes** [ocornut/imgui#1845]
ImGui 1.53

Hi, sorry to bother with old version issue, but from quick look it should still affect the current version as well: I'm getting FPU overflow exceptions, when I set 
```
io.MousePos = ImVec2(-FLT_MAX,-FLT_MAX);
```
when mouse cursor is out of the application window (like in the glfw_gl2 sample) and it got outside while dragging a ImGui window. It happens in ImGui::NewFrame (ImGui::UpdateMouseInputs() in current version) in mouse drag distance handling

```
        else if (g.IO.MouseDown[i])
        {
            ImVec2 mouse_delta = g.IO.MousePos - g.IO.MouseClickedPos[i];
            g.IO.MouseDragMaxDistanceAbs[i].x = ImMax(g.IO.MouseDragMaxDistanceAbs[i].x, mouse_delta.x < 0.0f ? -mouse_delta.x : mouse_delta.x);
            g.IO.MouseDragMaxDistanceAbs[i].y = ImMax(g.IO.MouseDragMaxDistanceAbs[i].y, mouse_delta.y < 0.0f ? -mouse_delta.y : mouse_delta.y);
            g.IO.MouseDragMaxDistanceSqr[i] = ImMax(g.IO.MouseDragMaxDistanceSqr[i], ImLengthSqr(mouse_delta));
}
```
If mouse_delta is ~FLT_MAX, ImLengthSqr causes overflow exception. I can work around it if I disable FPU exceptions, but I think that stopping the dragging at the last known cursor position would be more suitable.
Thanks for any help/suggestions!

**Comment by ocornut** [comment]
Hello David,

No worry for the "old version", this code imgui-side hasn't really changed and the issue makes sense.

> when mouse cursor is out of the application window (like in the glfw_gl2 sample) and it got outside while dragging a ImGui window. It happens in ImGui::NewFrame (ImGui::UpdateMouseInputs() in current version) in mouse drag distance handling

Which OS and GLFW version are you using?
I am still going to make an imgui side change to make this more sturdy on imgui side, but I think the problem lies in the back-end here. It is reporting a null MousePos while also reporting down buttons.

Normally when dragging from a click that happened in our window we should be capturing the mouse and allow valid mouse position outside of the window bounds. Therefore it is abnormal to have `io.MouseDown[0] = true` along with `io.MousePos = ImVec2(-FLT_MAX,-FLT_MAX);`

**Comment by dpethes** [comment]
I'm on Windows 7 with custom pascal impl on top of SDL2 2.0.8. 
Turns out you are right - it's a bug in my impl, ported from imgui_impl_sdl_gl2.cpp: I wasn't checking SDL_GetWindowFlags() result for SDL_WINDOW_INPUT_FOCUS when setting the io.MousePos, but for SDL_WINDOW_MOUSE_FOCUS instead. Seems this changed at some point after I ported it. I'm not on my desktop so I can't test it atm, but I'm quite sure that this is it - I'll test and report later.
Thanks for guiding me, have a nice day!

(note to self:  make sure to check the reference impl when upgrading to new version)
